### PR TITLE
fix(web-components): fixes back to top positioning

### PR DESCRIPTION
**Pull request by ad9242**
## Summary of the changes
Fixes positioning so that it no longer overlaps ic-footer

## Related issue
#197 

## Checklist
- [x] I have added relevant unit and visual regression tests.
- [x] I have manually accessibility tested any changes, if relevant.
- [x] I have ensured any changes match the Figma component library.

### CODE REVIEW

**Comment by GCHQ-Developer-530** [comment]
Now that it doesn't overlap the footer, is there an issue with overlapping the rest of the page content? e.g here I made the text longer and back to top overlaps the text
Screenshot 2023-01-03 at 14 13 28

File: packages/web-components/src/components/ic-back-to-top/ic-back-to-top.tsx [hunk 1]
```diff
@@ -42,7 +42,13 @@ export class BackToTop {
   };
 
   private setFooterVisible = (visible: boolean) => {
-    this.footerVisible = visible;
+    this.checkForClassificationBanner();
+    if (typeof window !== "undefined" && window.scrollY === 0) {
```
Comment (jd239) [inline packages/web-components/src/components/ic-back-to-top/ic-back-to-top.tsx:46]: Even though it's against our guidance, what happens if the footer is not below the fold but window.scrollY is 0?

**Review by jd239 — CHANGES_REQUESTED** [review]
Just a single comment! 😃

File: packages/web-components/src/components/ic-back-to-top/ic-back-to-top.tsx (hunk 1 above)
Comment (ad9242) [reply packages/web-components/src/components/ic-back-to-top/ic-back-to-top.tsx:46]: The back to top would not be displayed at all, as the target el would not have scrolled off screen yet

**Review by jd239 — APPROVED** [review]

**Comment by ad9242** [comment]
> Now that it doesn't overlap the footer, is there an issue with overlapping the rest of the page content? e.g here I made the text longer and back to top overlaps the text. Screenshot 2023-01-03 at 14 13 28 Not sure if there's a way for it to cause the text to wrap?

I think this is ok as the guidance shows the back to top overlaying content

**Review by GCHQ-Developer-530 — APPROVED** [review]

**Pull request merged**